c/tskit/genotypes.c

```diff
@@ -157,6 +141,17 @@ tsk_variant_init(tsk_variant_t *self, const tsk_treeseq_t *tree_sequence,
     tsk_size_t num_samples_alloc;
 
     tsk_memset(self, 0, sizeof(tsk_variant_t));
+    self->site = malloc(sizeof(*self->site));
```

Comment:
I think we could probably avoid this malloc, but let's not change it yet to keep the diff size managable. I've made  #2161 to track.